File: IT/Stringhe/Esercizi/Python/esercizio9.py
# file esercizio9.py
# -*- coding: utf-8 -*-

from __future__ import print_function, unicode_literals

import os

s = "prìncipi"
ss = ("Prìncipi", "princìpi")

def strEq(s1, s2):
	l = len(s1)
	if l != len(s2): return False

	for i in range(l):
		c = s1[i]
		c2 = s2[i]

		if c != c2 and c.lower() != c2.lower():
			return False
	return True

for t in ss:
	if strEq(s, t):
		print("Le parole \"{}\" e \"{}\" sono uguali.".format(s, t))
	else:
		print("Le parole \"{}\" e \"{}\" sono diverse.".format(s, t))

File: IT/Stringhe/Esercizi/C++/esercizio9.cpp
/*
file esercizio9.cpp
compilare con: g++ -std=c++11 esercizio9.cpp
*/

#include <iostream>
#include <sstream>

using std::wstring;
using std::wcout;
using std::endl;
using std::locale;

bool strEq(const wstring& s1, const wstring& s2, const locale& loc) {
	int l = s1.size();
	if (l != s2.size()) return false;

	for (int i = 0; i < l; i++) {
		wchar_t c = s1[i];
		wchar_t c2 = s2[i];

		if (c != c2 && tolower(c, loc) != tolower(c2, loc)) {
			return false;
		}
	}

	return true;
}

int main()
{
	setlocale(LC_ALL, "");
	locale loc("");

	wstring s = L"prìncipi";
	wstring ss[] = {L"Prìncipi", L"princìpi"};

	for (auto& t : ss) {
		if (strEq(s, t, loc)) {
			wcout << "Le parole \"" << s << "\" e \"" << t << "\" sono uguali." << endl;
		}
		else {
			wcout << "Le parole \"" << s << "\" e \"" << t << "\" sono diverse." << endl;
		}
	}
}

